### PR TITLE
Issue2446 fix data centers

### PR DESCRIPTION
**Pull request by khinkelman**
Hi @AntoineGautier, please find the revisions here to the Data Center package for issue #2446. Please let me know if you have any comments, questions, or if you need any further changes. Thanks!

### CODE REVIEW

**Comment by AntoineGautier** [comment]
The binding at https://github.com/lbl-srg/modelica-buildings/pull/2455/files#diff-96cc9a360941f39bd2c3f3da41dd4ede8741fce05fd048c64de5b8d674149aceR67
should be
`final dpFixed_nominal={if activate_ThrWayVal then dp2_nominal else 0,0},`
and at https://github.com/lbl-srg/modelica-buildings/pull/2455/files#diff-96cc9a360941f39bd2c3f3da41dd4ede8741fce05fd048c64de5b8d674149aceR107
`final dp2_nominal=if activate_ThrWayVal then 0 else dp2_nominal`

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialIntegratedPrimary.mo [hunk 1]
```diff
@@ -89,40 +89,50 @@ model PartialIntegratedPrimary
     final y_start=yVal6_start)
     "Bypass valve: closed when free cooling mode is deactivated;
     open when free cooling is activated"
-    annotation (Placement(transformation(extent={{-40,-30},{-60,-10}})));
+    annotation (Placement(transformation(extent={{-30,-30},{-50,-10}})));
 
+  Fluid.FixedResistances.Junction spl2(
+    redeclare package Medium = Medium2,
+    m_flow_nominal={numChi*m2_flow_chi_nominal,-numChi*m2_flow_chi_nominal,-
+        m2_flow_wse_nominal},
```
Comment (AntoineGautier) [inline Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialIntegratedPrimary.mo:97]: `-m2_flow_wse_nominal`

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialIntegratedPrimary.mo [hunk 2]
```diff
@@ -89,40 +89,50 @@ model PartialIntegratedPrimary
...
+    dp_nominal={0,0,0}) "Splitter"
+    annotation (Placement(transformation(extent={{90,-50},{70,-70}})));
+  Fluid.FixedResistances.Junction jun2(
+    redeclare package Medium = Medium2,
+    m_flow_nominal={m2_flow_wse_nominal,numChi*m2_flow_chi_nominal,-numChi*
+        m2_flow_chi_nominal},
```
Comment (AntoineGautier) [inline Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialIntegratedPrimary.mo:103]: `-numChi*m2_flow_chi_nominal, numChi*m2_flow_chi_nominal`

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimaryPlantSide.mo [hunk 3]
```diff
@@ -11,9 +11,30 @@ model IntegratedPrimaryPlantSide
...
+  Fluid.FixedResistances.Junction spl3(
+    redeclare package Medium = Medium2,
+    m_flow_nominal={numChi*m2_flow_chi_nominal,-m2_flow_wse_nominal,-numChi*
+        m2_flow_chi_nominal},
+    dp_nominal={0,0,0}) "Splitter"
+    annotation (Placement(transformation(extent={{0,-10},{-20,-30}})));
+  Fluid.FixedResistances.Junction jun3(
+    redeclare package Medium = Medium2,
+    m_flow_nominal={numChi*m2_flow_chi_nominal,-numChi*m2_flow_chi_nominal,-
+        m2_flow_wse_nominal},
```
Comment (AntoineGautier) [inline Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimaryPlantSide.mo:23]: `+m2_flow_wse_nominal`

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimarySecondary.mo [hunk 4]
```diff
@@ -140,41 +140,72 @@ model IntegratedPrimarySecondary
...
+  Fluid.FixedResistances.Junction spl2(
+    redeclare package Medium = Medium2,
+    m_flow_nominal={numChi*m2_flow_chi_nominal,-numChi*m2_flow_chi_nominal,-
+        m2_flow_wse_nominal},
+    dp_nominal={0,0,0}) "Splitter"
+    annotation (Placement(transformation(extent={{90,-50},{70,-70}})));
+  Fluid.FixedResistances.Junction jun3(
+    redeclare package Medium = Medium2,
+    m_flow_nominal={numChi*m2_flow_chi_nominal,m2_flow_wse_nominal,-numChi*
+        m2_flow_chi_nominal},
```
Comment (AntoineGautier) [inline Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimarySecondary.mo:162]: Port 3 connected to WSE leg

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimarySecondary.mo [hunk 5]
```diff
@@ -140,41 +140,72 @@ model IntegratedPrimarySecondary
...
     "Constant speed pumps"
-    annotation (Placement(transformation(extent={{10,-30},{-10,-10}})));
+    annotation (Placement(transformation(extent={{-20,-30},{-40,-10}})));
   Buildings.Fluid.Sensors.MassFlowRate bypFlo(redeclare package Medium = Medium2)
     "Bypass water mass flowrate"
     annotation (Placement(transformation(extent={{-40,-70},{-20,-50}})));
+  Fluid.FixedResistances.Junction jun2(
+    redeclare package Medium = Medium2,
+    m_flow_nominal={m2_flow_wse_nominal,numChi*m2_flow_chi_nominal,-numChi*
+        m2_flow_chi_nominal},
```
Comment (AntoineGautier) [inline Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimarySecondary.mo:150]: `-numChi*m2_flow_chi_nominal, numChi*m2_flow_chi_nominal`

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/NonIntegrated.mo [hunk 6]
```diff
@@ -10,19 +10,31 @@ model NonIntegrated
...
+  Fluid.FixedResistances.Junction spl2(
+    redeclare package Medium = Medium2,
+    m_flow_nominal=
+      {numChi*m2_flow_chi_nominal,-numChi*m2_flow_chi_nominal,-m2_flow_wse_nominal},
+    dp_nominal={0,0,0}) "Splitter"
+    annotation (Placement(transformation(extent={{90,-50},{70,-70}})));
+  Fluid.FixedResistances.Junction jun2(
+    redeclare package Medium = Medium2,
+    m_flow_nominal=
+      {m2_flow_wse_nominal,numChi*m2_flow_chi_nominal,-numChi*m2_flow_chi_nominal},
```
Comment (AntoineGautier) [inline Buildings/Applications/DataCenters/ChillerCooled/Equipment/NonIntegrated.mo:22]: `-numChi*m2_flow_chi_nominal, +numChi*m2_flow_chi_nominal`

**Review by AntoineGautier — CHANGES_REQUESTED** [review]
@khinkelman Thanks for the updates. Please see my inline comments and https://github.com/lbl-srg/modelica-buildings/pull/2455#issuecomment-823618321.

**Comment by AntoineGautier** [comment]
@khinkelman I just merged your commits and one of mine (typos) but it may be confusing. Please still consider my review comments to be addressed on your fork.

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/WatersideEconomizer.mo [hunk 7]
```diff
@@ -2,9 +2,9 @@ within Buildings.Applications.DataCenters.ChillerCooled.Equipment;
 model WatersideEconomizer "Waterside economizer"
   extends Buildings.Applications.DataCenters.ChillerCooled.Equipment.BaseClasses.PartialPlantParallel(
     final num=1,
-    val2(each final dpFixed_nominal=0),
+    val2(each final dpFixed_nominal=dp2_nominal),
     val1(each final dpFixed_nominal=dp1_nominal),
-    kFixed={m1_flow_nominal/sqrt(dp1_nominal),0},
+    kFixed={m1_flow_nominal,m2_flow_nominal} ./ sqrt({dp1_nominal,dp2_nominal}),
```
[the remark this answers is not in the document]
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/WatersideEconomizer.mo:7]: Made `kFixed` final

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialChillerWSE.mo [hunk 8]
```diff
@@ -48,7 +48,7 @@ partial model PartialChillerWSE
     "Valve leakage, l=Kv(y=0)/Kv(y=1)"
     annotation(Dialog(group="Two-way valve"));
   parameter Real[2] kFixedValWSE(each unit="", each min=0)=
-    {m1_flow_wse_nominal/ sqrt(dp1_wse_nominal),0}
+    {m1_flow_wse_nominal,m2_flow_wse_nominal} ./ sqrt({dp1_wse_nominal,dp2_wse_nominal})
```
[the remark this answers is not in the document]
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialChillerWSE.mo:51]: Deleted parameter `kFixedValWSE` from this class, as `kFixed` was made final in `Buildings.Applications.DataCenters.ChillerCooled.Equipment.WatersideEconomizer`.

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/WatersideEconomizer.mo [hunk 9]
```diff
@@ -2,9 +2,9 @@ within Buildings.Applications.DataCenters.ChillerCooled.Equipment;
 model WatersideEconomizer "Waterside economizer"
```
[the remark this answers is not in the document]
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/WatersideEconomizer.mo:2]: Added parameter `dpThrWayVal_nominal` to base class `Buildings.Applications.DataCenters.ChillerCooled.Equipment.BaseClasses.ThreeWayValveParameters` (which this `WatersideEconomizer` model extends).

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialChillerWSE.mo [hunk 10]
```diff
@@ -269,6 +269,18 @@ partial model PartialChillerWSE
     "Temperature sensor"
     annotation (Placement(transformation(extent={{28,14},{8,34}})));
 
+  Fluid.FixedResistances.Junction spl1(
+    redeclare package Medium = Medium1,
+    m_flow_nominal={numChi*m1_flow_chi_nominal,-numChi*m1_flow_chi_nominal,-
+        m1_flow_wse_nominal},
```
[the remark this answers is not in the document]
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialChillerWSE.mo:275]: Done

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialChillerWSE.mo [hunk 11]
```diff
@@ -269,6 +269,18 @@ partial model PartialChillerWSE
...
+    dp_nominal={0,0,0}) "Splitter"
+    annotation (Placement(transformation(extent={{-80,46},{-60,66}})));
+  Fluid.FixedResistances.Junction jun1(
+    redeclare package Medium = Medium1,
+    m_flow_nominal={numChi*m1_flow_chi_nominal,m1_flow_wse_nominal,-numChi*
+        m1_flow_chi_nominal},
```
[the remark this answers is not in the document]
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialChillerWSE.mo:281]: Done

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialIntegratedPrimary.mo (hunk 1 above)
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialIntegratedPrimary.mo:97]: Negative sign is there (it is just on the line above). Formatting fixed for better clarity.

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialIntegratedPrimary.mo (hunk 2 above)
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/BaseClasses/PartialIntegratedPrimary.mo:103]: Done

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimaryPlantSide.mo (hunk 3 above)
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimaryPlantSide.mo:23]: Done

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimarySecondary.mo (hunk 5 above)
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimarySecondary.mo:150]: Done

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimarySecondary.mo (hunk 4 above)
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/IntegratedPrimarySecondary.mo:162]: Done

File: Buildings/Applications/DataCenters/ChillerCooled/Equipment/NonIntegrated.mo (hunk 6 above)
Comment (khinkelman) [reply Buildings/Applications/DataCenters/ChillerCooled/Equipment/NonIntegrated.mo:22]: Done

**Comment by khinkelman** [comment]
Hi @AntoineGautier thank you for your review and comments. I addressed the inline comments, but it unclear to me where the changes should be made with [#2455 (comment)](https://github.com/lbl-srg/modelica-buildings/pull/2455#issuecomment-823618321). The links don't seem to go to the correct locations. Could you please clarify what changes you would like and where? 

Thanks!
Katy

**Comment by AntoineGautier** [comment]
@khinkelman Thanks for your updates. Sorry for the broken links (I inadvertently merged your PR before you had the chance to address the review comments). I edited https://github.com/lbl-srg/modelica-buildings/pull/2455#issuecomment-823618321 with the class names (instead of GH links) for clarity.

**Comment by khinkelman** [comment]
Thanks @AntoineGautier! One follow up question on your first bullet point:

In `Buildings.Applications.DataCenters.ChillerCooled.Equipment.BaseClasses.PartialHeatExchanger`, the instance of `thrWayVal` has the conditional statement `if activate_ThrWayVal`. Thus, I wonder if the internal parameter `pFixed_nominal` also needs the conditional statement you proposed, if the entire valve is removed when `activate_ThrWayVal = false`?

**Comment by khinkelman** [comment]
- [ ] Note to self: Verify unit tests are not affected by new changes.

**Comment by AntoineGautier** [comment]
> Thanks @AntoineGautier! One follow up question on your first bullet point:
> 
> In `Buildings.Applications.DataCenters.ChillerCooled.Equipment.BaseClasses.PartialHeatExchanger`, the instance of `thrWayVal` has the conditional statement `if activate_ThrWayVal`. Thus, I wonder if the internal parameter `pFixed_nominal` also needs the conditional statement you proposed, if the entire valve is removed when `activate_ThrWayVal = false`?

You are right, I miss that. The conditional instance is enough, no need for a conditional binding.

**Comment by khinkelman** [comment]
Sounds great, thanks! I'll create a new PR then with the updates per your comments.

**Pull request merged**